maven-core/src/main/java/org/apache/maven/lifecycle/internal/MojoExecutor.java

```diff
@@ -232,7 +237,17 @@ private static class ProjectLock implements AutoCloseable
                 boolean aggregator = mojoDescriptor.isAggregator();
                 acquiredAggregatorLock = aggregator ? aggregatorLock.writeLock() : aggregatorLock.readLock();
                 acquiredProjectLock = getProjectLock( session );
-                acquiredAggregatorLock.lock();
+                if ( !acquiredAggregatorLock.tryLock() )
+                {
+                    for ( String s : MessageHelper.formatWarning(
```

Comment:
An aggregator mojo is already being executed in this parallel build, those kind of mojos require...

..until the aggregator mojo is done.